pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/DistinctFromTransformFunction.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.operator.transform.function;
+
+import java.util.Arrays;
+import java.util.BitSet;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.pinot.common.function.TransformFunctionType;
+import org.apache.pinot.core.operator.blocks.ProjectionBlock;
+import org.apache.pinot.core.operator.transform.TransformResultMetadata;
+import org.apache.pinot.segment.spi.datasource.DataSource;
+import org.apache.pinot.segment.spi.index.reader.NullValueVectorReader;
+import org.roaringbitmap.PeekableIntIterator;
+
+
+/**
+ * <code>DistinctFromTransformFunction</code> abstracts the transform needed for IsDistinctFrom and IsNotDistinctFrom.
+ * Null value is considered as distinct from non-null value.
+ * When both values are not null, this function calls equal transform function to determined whether two values are
+ * distinct.
+ * This function only supports two arguments which are both column names.
+ */
+public class DistinctFromTransformFunction extends BinaryOperatorTransformFunction {
+  private int[] _results;
+  // null iterator for left argument.
+  private PeekableIntIterator _leftNullValueVectorIterator;
+  // null iterator for right argument.
+  private PeekableIntIterator _rightNullValueVectorIterator;
+  // If _getDistinct is 1, act IsDistinctFrom transformation, otherwise act as IsNotDistinctFrom.
+  private int _getDistinct;
+
+  // getDistinct is set to true for IsDistinctFrom, otherwise it is for IsNotDistinctFrom.
+  protected DistinctFromTransformFunction(boolean getDistinct) {
+    super(getDistinct ? TransformFunctionType.NOT_EQUALS : TransformFunctionType.EQUALS);
+    _getDistinct = getDistinct ? 1 : 0;
+  }
+
+  // Returns a bit vector of document length where each bit represents whether the row is null or not.
+  // Set the value as not null by default if null option is disabled.
+  static private BitSet getNullBits(@Nullable PeekableIntIterator nullValueVectorIterator, int length, int[] docIds) {
```

Comment:
(convention) We usually put `private` in front of `static`. Same for other places

Reply:
Done.